api/v1beta1/gitrepository_types.go

```diff
@@ -87,6 +87,15 @@ type GitRepositorySpec struct {
 	// This option is available only when using the 'go-git' GitImplementation.
 	// +optional
 	RecurseSubmodules bool `json:"recurseSubmodules,omitempty"`
+
+	// Extra repositories to map into the repository
+	Include []GitRepositoryInclude `json:"include,omitempty"`
+}
+
+//
+type GitRepositoryInclude struct {
+	GitRepository meta.NamespacedObjectReference `json:"repository"`
+	Path          string                         `json:"path"`
```

Comment:
Should probably change this to a from and to path.

Reply:
Conventionally, `From` could default to the root of the dependency GitRepository while `To` can take the name of the GitRepository resource as default.